apps/address-service/domains/common/utils/services/augmentAddressSuggestions.js

```diff
@@ -0,0 +1,81 @@
+const { AddressSource } = require('@address-service/domains/address/utils/serverSchema')
+const { BUILDING_ADDRESS_TYPE } = require('@address-service/domains/common/constants/addressTypes')
+
+/**
+ * Augments the suggestions list with data from AddressSource.
+ * * This function takes an initial list of address suggestions (e.g., from Google Places API)
+ * and enriches it with matching addresses found in the local AddressSource.
+ * It's designed to solve the issue of missing custom-added addresses in the search results.
+ *
+ * @see https://github.com/open-condo-software/condo/issues/6485
+ * @param {object} context - The context required to execute Keystone queries.
+ * @param {Array<NormalizedSuggestion>} suggestions - The normalized array of suggestions from AddressProvider.
+ * @returns {Promise<Array<object>>} A new array containing both original and augmented suggestions.
+ */
+
+async function getAugmentedSuggestions (context, suggestions) {
```

Comment:
When I see "getAugmentedSuggestions," I think this function will return only augmented suggestions, and I do not understand why it needs suggestions found before.
What do you think about renaming the function to "pushAugmentedSuggestions"? Or something else to make the reader understand what this function really does.